### ISSUE TITLE
Documentation probably advises unlicensed use of Visual Studio Build Tools

### ISSUE THREAD

**Issue by max630** [microsoft/vscode#95745]
- VSCode Version: None
- OS Version: None

Steps to Reproduce:

1. Read https://code.visualstudio.com/docs/cpp/config-msvc instruction to install build tools
2. Follow the link "https://visualstudio.microsoft.com/downloads/#other", download installer from "Build Tools for Visual Studio 2019"
3. From the installer, open the license terms https://visualstudio.microsoft.com/license-terms/mlt031519/

The license has the text:

> INSTALLATION AND USE RIGHTS.
> You may install and use any number of copies of the software **to use solely with Visual Studio Community, Visual Studio Professional, and Visual Studio Enterprise**, to develop and test your applications.

Does this license really allow using the tools with Visual Studio Core? Reading further, it seems like it is only intended be used for CI builds.

I think you should contact to update the tools license, or state in documentation that standalone tools are not supposed to be used with VS Code

Does this issue occur when all extensions are disabled?: Yes

**Comment by chrisdias** [comment]
this is an issue on the c++ extension https://github.com/microsoft/vscode-cpptools

**Comment by mluparu** [comment]
**Tl;dr: You can use the C++ Toolset from Visual Studio Build Tools along with Visual Studio Code** to compile, build, and verify any C++ codebases as long as you also have a valid Visual Studio license (either Community, Pro or Enterprise) that you are actively using to develop these C++ codebases.

More details: 

- Visual Studio license is a named license, which means that if you have acquired such a license, you can install the product on as many devices as you want, as long as you’re the sole user of the product. 
- Visual Studio Build Tools ships as a supplement to Visual Studio and allows you to install and use it to compile, build, and verify with your codebases developed with Visual Studio rather than using the full Visual Studio product for these activities. Build Tools does require you have a valid Visual Studio license (Community, Professional or Enterprise)
- If you also have Visual Studio Code installed on any devices where Build Tools is installed, then, you are able to use the C++ Toolset from the Visual Studio Build Tools along with Visual Studio Code for compiling/building/verifying the C++ codebases created with Visual Studio. Configuration instructions are outlined in this article https://code.visualstudio.com/docs/cpp/config-msvc 
 
In conclusion, if you own a Visual Studio license and used it with Build Tools to build a C++ codebase, you can use the C++ toolset in the Visual Studio Build Tools from your Visual Studio Code editor for the purpose of compiling/building/verifying your C++ codebases without having to install the full Visual Studio IDE on that machine. Of course, Visual Studio Build Tools is designed primarily for CI environments, where it isn’t practical to install the full Visual Studio IDE. For any other scenarios, including active development, we recommend using Visual Studio (which includes all tools in Visual Studio Build Tools) instead of the Visual Studio Code + Visual Studio Build Tools combination. 

We will update the article mentioned above to add this clarification as well.

**Comment by chrisdias** [comment]
i think @mluparu answered this, closing.
